tests/rptest/services/datalake/util/table_verifier.py

```diff
@@ -0,0 +1,87 @@
+import json
+import logging
+
+
+def safe_decode(val):
+    if isinstance(val, bytes):
+        return val.decode("utf-8", errors="replace")
+    if isinstance(val, memoryview):
+        return val.tobytes().decode("utf-8", errors="replace")
+    return str(val) if val is not None else ""
+
+
+def canonical_json(val):
+    """Serialize dicts to canonical JSON strings for comparison."""
```

Comment:
This method doesn't do what the name and comment says. It returns strings only for dicts and doesn't do anything for all the other input types.